infra/lambda/datasets-endpoint/test_function.py

```diff
@@ -0,0 +1,307 @@
+"""
+Basic Lambda function tests. Working Data Lake AWS environment is required
+(run '$ cdk deploy' before running tests).
+"""
+
+import logging
+import re
+import uuid
+
+import function  # pylint:disable=import-error
+import pytest
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def test_request_no_method():
+    """Try request not containing method."""
+
+    body = {}
+
+    resp = function.lambda_handler({"body": body}, "context")
+    logger.info(resp)
+
+    assert resp["statusCode"] == 400
+    assert resp["body"]["message"] == "Bad Request: 'httpMethod' is a required property."
+
+
+def test_request_no_body():
+    """Try request not containing body."""
+
+    method = "POST"
+
+    resp = function.lambda_handler({"httpMethod": method}, "context")
+    logger.info(resp)
+
+    assert resp["statusCode"] == 400
+    assert resp["body"]["message"] == "Bad Request: 'body' is a required property."
+
+
+def test_post_method(db_truncate):  # pylint:disable=unused-argument
+    """Test Dataset creation using POST method."""
+
+    method = "POST"
+    body = {}
+    body["type"] = "RASTER"
+    body["title"] = "Dataset {}".format(uuid.uuid4().hex[:8])
+    body["owning_group"] = "A_XYZ_XYZ"
+
+    resp = function.lambda_handler({"httpMethod": method, "body": body}, "context")
+    logger.info(resp)
+
+    assert resp["statusCode"] == 201
+    assert resp["body"]["type"] == body["type"]
+    assert resp["body"]["title"] == body["title"]
+    assert resp["body"]["owning_group"] == body["owning_group"]
+
+    pytest.dataset_id = resp["body"]["id"]
+    pytest.dataset_type = resp["body"]["type"]
+    pytest.dataset_title = resp["body"]["title"]
+    pytest.owning_group = resp["body"]["owning_group"]
+
+
+def test_post_method_missing_attr():
```

Comment:
This is testing third party code, the schema validator. We can't change the schema validator, so we should instead be testing our code. In this case, that might mean simply testing the schema itself, verifying that a certain call ends up being validated by a specific schema. To do that we might create a method which takes the request as its input and returns the relevant schema. We can then test the method without ever calling out to Lambda.

(Of course, we'd also have at least some tests which *do* call out to Lambda to create instances, to verify that we can actually create instances. Basically the idea is to separate functional code, such as validation, so that it can be tested quickly and therefore thoroughly using unit tests.)

Reply:
@l0b0 , I understand what you mean, but I don't have any good idea how to implement it in consistency with other code. Do you have any suggestion or we can leave it as is for now ?

Reply:
I'm not familiar enough with the rest of the code to tell how it would be inconsistent. But the general idea would be to call a function like `get_body_schema` in `lambda_handler` to get the body schema and pass that into the method handler. We can then unit test `get_body_schema`, avoiding any Lambda or DynamoDB setup.